systemd-hosts.d.service

```diff
@@ -3,13 +3,66 @@ Description=Keep your hosts records in logically separated files
 After=local-fs.target
 
 [Service]
+# Security
+ProtectSystem=yes
+ProtectHome=yes
+PrivateDevices=yes
+ProtectKernelTunables=yes
+ProtectKernelModules=yes
+ProtectControlGroups=yes
+ProtectKernelLogs=yes
+SystemCallFilter=@system-service
+SystemCallErrorNumber=EPERM
+NoNewPrivileges=yes
+PrivateTmp=yes
+LockPersonality=yes
+MemoryDenyWriteExecute=yes
+ProtectClock=yes
+RestrictNamespaces=yes
+RestrictRealtime=yes
+RestrictSUIDSGID=yes
+RuntimeDirectory=systemd/hosts.d
+RuntimeDirectoryPreserve=yes
+
+# Service
 Type=oneshot
+
+# Environment
 Environment="HOSTS_FILE=/etc/hosts"
 Environment="HOSTS_DIRECTORY=/etc/hosts.d"
-ExecStart=/bin/sh -c 'echo -e "## Generated by systemd-hosts.d ##\\n" > $HOSTS_FILE'
-ExecStart=/bin/sh -c 'echo -e "## All changes in this file will be ignored ##\\n" >> $HOSTS_FILE'
-ExecStart=/bin/sh -c 'cat $HOSTS_DIRECTORY/*.conf | grep \'^[[:blank:]]*[^[:blank:]#;]\' >> $HOSTS_FILE'
-ExecStart=/bin/chmod ugo=r $HOSTS_FILE
+# Loosely match IPv4 or IPv6 followed by hostname and hostalias(s)
+Environment="HOSTS_REGEX_BASE=^\\([0-9]\\{1,3\\}\\.\\|[0-9a-fA-F]\\{1,4\\}:\\|::\\).\\+\\?\\(\\s\\+[a-zA-Z0-9.-]\\{1,\\}\\)\\{1,\\}$"
```

Comment:
This is a quite hard-to-understand expression. Fwik hosts file parsers are plenty capable of ignoring invalid lines by themselves. So I don't think there is any benefit of doing so ourselves, is there? Feel free to remove the filtering step altogether.

For the clarity, the original filter's indented purpose was literally to "save on space/speed up parsing", which doesn't seem  to me to hold true, especially knowing that I'd probably spent more time thinking about that, than it actually saved anything in a meaningful way.

Reply:
The benefit is in to gain some level of confidence that invalid lines are filtered out of final hosts file. It's hard to understand due to all of the excessive escape slashing 😆